File: cpp/223.cpp
using ll = long long;
class Solution {
public:
  int computeArea(ll A, ll B, ll C, ll D, ll E, ll F, ll G, ll H) {
    ll total = (C-A)*(D-B) + (G-E)*(H-F);
    ll w = min(C, G) - max(A, E);
    if(w < 0) w = 0;
    ll h = min(H, D) - max(B, F);
    if(h < 0) h = 0;
    return (int)(total - h*w);
  }
};

File: python/223.py
class Solution(object):
    def computeArea(self, A, B, C, D, E, F, G, H):
        total = (C-A)*(D-B) + (G-E)*(H-F)
        w = min(C, G) - max(A, E)
        if w < 0:
            w = 0
        h = min(H, D) - max(B, F)
        if h < 0:
            h = 0
        return total - h*w
